### ISSUE TITLE
鸿蒙Next纯血版本如何接入mmkv？

### ISSUE THREAD

**Issue by ChawLoo** [Tencent/MMKV#1261]
原先想通过编译成CPP引入，但是那一块不太熟悉，原先打算用CMake编译后去接入，但是搞不明白

**Comment by lingol** [comment]
The official support of MMKV for OpenHarmonyOS is under construction. You can wait for it.

**Comment by ChawLoo** [comment]
预估多久面市？

**Comment by lingol** [comment]
I can't say for sure. Eventually, it should be released before the next big event of the OHOS Next.

**Comment by alhah** [comment]
Does the big event mean WeChat porting to OHOS?

**Comment by lingol** [comment]
The undergoing work can be found in the branch `dev_openharmony`. It's still in an early stage.

**Comment by wangyatong** [comment]
24年年底之前绝对上线🤪

**Comment by lingol** [comment]
For those who would try, it's live on OHPM. https://ohpm.openharmony.cn/#/cn/detail/@tencent%2Fmmkv
wiki https://github.com/Tencent/MMKV/wiki/ohos_setup

**Comment by ChawLoo** [comment]
牛啊牛啊

**Comment by ChawLoo** [comment]
最新测试结果，真机一样有问题

**Comment by pioneerz** [comment]
> 最新测试结果，真机一样有问题



> 我mini m2芯片，跑的模拟器 ![image](https://private-user-images.githubusercontent.com/26214519/323506158-f70b2c48-516b-49fc-9d16-9fdfe8f24d0c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM0MjgzMzcsIm5iZiI6MTcxMzQyODAzNywicGF0aCI6Ii8yNjIxNDUxOS8zMjM1MDYxNTgtZjcwYjJjNDgtNTE2Yi00OWZjLTlkMTYtOWZkZmU4ZjI0ZDBjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE4VDA4MTM1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE0NzA5OGU4ODJkMjRhNzU3YTkwMDg1ZjdjYThjOTZlYjlkOTI2YzUzMGFkNGY2MzkzMGQwNzc4ODk2NTA2YzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.9P9X2GjG6lLW3SQGAohXU_n0YRlS6gUTNaLQx2OlOLk)

![image](https://github.com/Tencent/MMKV/assets/18280399/b513d26a-153b-4af6-96bb-2c66a2ba0fdd)
这个限制还挺多的

**Comment by ChawLoo** [comment]
> > 最新测试结果，真机一样有问题
> 
> > 我mini m2芯片，跑的模拟器 ![image](https://private-user-images.githubusercontent.com/26214519/323506158-f70b2c48-516b-49fc-9d16-9fdfe8f24d0c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM0MjgzMzcsIm5iZiI6MTcxMzQyODAzNywicGF0aCI6Ii8yNjIxNDUxOS8zMjM1MDYxNTgtZjcwYjJjNDgtNTE2Yi00OWZjLTlkMTYtOWZkZmU4ZjI0ZDBjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE4VDA4MTM1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE0NzA5OGU4ODJkMjRhNzU3YTkwMDg1ZjdjYThjOTZlYjlkOTI2YzUzMGFkNGY2MzkzMGQwNzc4ODk2NTA2YzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.9P9X2GjG6lLW3SQGAohXU_n0YRlS6gUTNaLQx2OlOLk)
> 
> ![image](https://private-user-images.githubusercontent.com/18280399/323521428-b513d26a-153b-4af6-96bb-2c66a2ba0fdd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM0MzA5MDEsIm5iZiI6MTcxMzQzMDYwMSwicGF0aCI6Ii8xODI4MDM5OS8zMjM1MjE0MjgtYjUxM2QyNmEtMTUzYi00YWY2LTk2YmItMmM2NmEyYmEwZmRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDE4VDA4NTY0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyYmZkMWRkZjBlMmVkN2MyMTlmMDlhZTI4N2NkZThkNDAzNTExY2NmMzI5ZTk1NDJmNjBjZTNkOWMyZjc0YWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.DWIb7hAo-A1Hl9a1BMsWnmTNXyrHkondD3wWag22zFU) 这个限制还挺多的

但是我满足呀，手机目前Mate 60 Pro Preview 2版本 ，模拟器也是Beta1版本
![image](https://github.com/Tencent/MMKV/assets/26214519/f8df132a-bb96-40c1-921a-70f0b4f7d67c)

**Comment by lingol** [comment]
What's the log from hilog?

**Comment by ChawLoo** [comment]
> What's the log from hilog?

04-19 09:06:29.479   43415-43415  A03D00/JSAPP                   pid-43415             I     Callee constructor is OK string
04-19 09:06:29.479   43415-43415  A03D00/JSAPP                   pid-43415             I     Ability::constructor callee is object [object Object]
04-19 09:06:29.487   43415-43415  A00000/MMKV                    pid-43415             I     <native_bridge.cpp:337::initialize> NAPI Error: code 6, msg Need number
04-19 09:06:29.499   43415-43415  A00000/testTag                 pid-43415             I     Ability onWindowStageCreate
04-19 09:06:29.499   43415-43415  A03D00/JSAPP                   pid-43415             I     初始化配置缓存框架
04-19 09:06:29.501   43415-43415  A00000/testTag                 pid-43415             I     Ability onForeground


![image](https://github.com/Tencent/MMKV/assets/26214519/cb158947-79b5-45c6-8012-b314729da7e4)

**Comment by lingol** [comment]
I see. There's a bug in `MMKV.initialize()` on handling optional parameters. Will fix in the next version. 
For the time being, you can bypass this by passing the `logLevel` param:
```js
import { MMKV, MMKVLogLevel } from '@tencent/mmkv';

let appCtx = this.context.getApplicationContext();
MMKV.initialize(appCtx, MMKVLogLevel.LevelInfo);
```

**Comment by lingol** [comment]
It's fixed in "@tencent/mmkv": "1.3.5-alpha".

**Comment by lebusishu** [comment]
> 我mini m2芯片，跑的模拟器、真机跑了也有问题 ![image](https://private-user-images.githubusercontent.com/26214519/323506158-f70b2c48-516b-49fc-9d16-9fdfe8f24d0c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NTUyNDksIm5iZiI6MTcxMzc1NDk0OSwicGF0aCI6Ii8yNjIxNDUxOS8zMjM1MDYxNTgtZjcwYjJjNDgtNTE2Yi00OWZjLTlkMTYtOWZkZmU4ZjI0ZDBjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDIyVDAzMDIyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgxNmYxMjZiN2VjY2IzMzJmMmQ0Nzc0Mjc5YThmNTAzOWZlMzUzNDNlYjYyNzgxMGMwYTA2NTI0ZDA4ZTI3MjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-eWawB2q2_yFsfA7_dbkZ-VxNqUmng7e-6JK24DMwHk)

也遇到了，需要指定MMKVLogLevel才可以，我这样就可以了：MMKV.initializeWithPath(filePath, context.cacheDir, MMKVLogLevel.LevelInfo)

**Comment by lingol** [comment]
> It's fixed in "@tencent/mmkv": "1.3.5-alpha".


It's fixed in "@tencent/mmkv": "1.3.5-alpha".

Try it.

**Comment by lingol** [comment]
> > 我mini m2芯片，跑的模拟器、真机跑了也有问题 ![image](https://private-user-images.githubusercontent.com/26214519/323506158-f70b2c48-516b-49fc-9d16-9fdfe8f24d0c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTM3NTUyNDksIm5iZiI6MTcxMzc1NDk0OSwicGF0aCI6Ii8yNjIxNDUxOS8zMjM1MDYxNTgtZjcwYjJjNDgtNTE2Yi00OWZjLTlkMTYtOWZkZmU4ZjI0ZDBjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDIyVDAzMDIyOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgxNmYxMjZiN2VjY2IzMzJmMmQ0Nzc0Mjc5YThmNTAzOWZlMzUzNDNlYjYyNzgxMGMwYTA2NTI0ZDA4ZTI3MjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-eWawB2q2_yFsfA7_dbkZ-VxNqUmng7e-6JK24DMwHk)
> 
> 也遇到了，需要指定MMKVLogLevel才可以，我这样就可以了：MMKV.initializeWithPath(filePath, context.cacheDir, MMKVLogLevel.LevelInfo)

Try it. 
It's fixed in "@tencent/mmkv": "1.3.5-alpha".

**Comment by lingol** [comment]
It's released in [v1.3.5](https://github.com/Tencent/MMKV/releases/tag/v1.3.5).
